libvirt/domain.go

```diff
@@ -237,15 +237,18 @@ func setCoreOSIgnition(d *schema.ResourceData, domainDef *libvirtxml.Domain) err
 			return err
 		}
 
-		domainDef.QEMUCommandline = &libvirtxml.DomainQEMUCommandline{
-			Args: []libvirtxml.DomainQEMUCommandlineArg{
-				{
-					Value: "-fw_cfg",
-				},
-				{
-					Value: fmt.Sprintf("name=opt/com.coreos/config,file=%s", ignitionKey),
+		if fwCfg, ok := d.GetOk("fw_cfg_name"); ok {
```

Comment:
we can add also here a comment to the cryptic name.
``fw_cfg` stands for firmware config is defined by a key and a value.`

Another thing, but this will be out of scope: consider that your create the disk and so on with the example.

Now we need to think about the `update` operation. the update operation is when you have a terraform infra already deployed (`terraform apply`, and we change the tfstate file, e.g one change the name.

To handle this operation is kind tricky for many reasons, but I would like that we could think about this.. In some resources this update can't be implemented and we have already in codebase some incongruence with this. However we might think.

Reply:
IMO it was handled by the Terraform Option 
```go
ForceNew: true
```
If `fw_cfg_name` is updated, actions after will be created from scratch no ?

Reply:
ah right anyways it is more tricky in other places but for this looks good resolving thx